qol: управление замком airalarm через Alt-клик

vovakons opened the pull request:
## Описание

Alt+клик по аир алярму теперь работает также как у АПЦ.

## Причина создания ПР / Почему это хорошо для игры

Удобство атмосам, не надо больше доставать карту из кошелька тратя на это время.

vovakons committed:
qol: alt click toggle lock for air alarms

vovakons committed:
refactor code

littleboobs reviewed (changes requested):
Наконец-то кто-то это сделал, красава. Видео бы, что всё работает предоставить. А ещё не уверен, но Эшель скорее всего заценит, замену ретюрнов на элс и элс ифы :)

vovakons commented:
> Наконец-то кто-то это сделал, красава. Видео бы, что всё работает предоставить. А ещё не уверен, но Эшель скорее всего заценит, замену ретюрнов на элс и элс ифы :)

Проверил на локалке, с картой даже внутри пда работает. Код скопирован у кода апц и немного с самого аиралярма, появилось немного дублирования кода, но не стал лезть в старую функцию чтобы ничего не ломать.

dageavtobusnick commented on code/game/machinery/alarm.dm:
```diff
@@ -1119,6 +1119,26 @@
+	else if(allowed(user) && !wires.is_cut(WIRE_IDSCAN))
+		locked = !locked
+		to_chat(user, span_notice("You [ locked ? "lock" : "unlock"] the Air Alarm interface."))
+	else
```
Разбей с помощью ранних ретурнов, как это сделано ниже

dageavtobusnick replied:
Так же избавься от копирования кода тут и в attackby, перенеся функционал начиная от add_fingerprint  и заканчивая to_chat(user, span_warning("Access denied.")) оттуда в этот прок. Страх сломать - не повод плодить копирования.

dageavtobusnick commented on code/game/machinery/alarm.dm:
```diff
@@ -1119,6 +1119,26 @@
+	var/obj/item/card/id/card = user.get_id_card()
+	if(!istype(card))
+		return NONE
+	add_fingerprint(user)
```
Дублирование тут и в togglelock. Оставь только в togglelock.

vovakons replied:
Исправил

vovakons committed:
fixes after review

dageavtobusnick commented on code/game/machinery/alarm.dm:
```diff
@@ -1119,6 +1119,26 @@
+		return
+	if(allowed(user) && !wires.is_cut(WIRE_IDSCAN))
+		locked = !locked
+		to_chat(user, span_notice("You [ locked ? "lock" : "unlock"] the Air Alarm interface."))
```
Тут потерял SStgui.update_uis(src)

dageavtobusnick commented on code/game/machinery/alarm.dm:
```diff
@@ -1119,6 +1119,26 @@
 			if(wiresexposed)
 				. += span_notice("The wiring could be <i>cut and removed</i> or panel could <b>screwed</b> closed.")
 
+/obj/machinery/alarm/proc/togglelock(mob/living/user)
```
Замени на вызов этого прока код с идентичным содержимым в attackby при выполнении условия if(I.GetID() || is_pda(I)), оставив там лишь вызов прока и return ATTACK_CHAIN_PROCEED.

vovakons committed:
fix after review

Добавлен вызов SStgui.update_uis внутри togglelock.
В proc attackby добавлен вызов togglelock вместо дублирования кода.
